.pre-commit-config.yaml

```diff
@@ -0,0 +1,10 @@
+default_install_hook_types: [pre-commit, pre-push]
+
+repos:
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.12.9
+    hooks:
+      # Run the linter, applying any available fixes
+      - id: ruff-check
+        stages: [ pre-push ]
```

Comment:
This only triggers on push. I think the pre-commit stage would also be useful. Side note, I pushed the commit https://github.com/SasView/sasview/commit/ad11516c6ecc4a1f18a77d2ad49a7eee4c1188c0, which has a bad indent and no blank line at the end of the file, was able to be pushed without issue.

Things we will need for this:
- A setup procedure (pre-commit install, I believe is all that is needed)
- A list of checks pre-commit will attempt to correct
- Triggering pre-commit and what happens if a fix is found